### ISSUE TITLE
`pip` completion used for `pipx` command

### ISSUE THREAD

**Issue by sevens-ef** [xonsh/xonsh#3526]
I have `pipx` installed and its bash completion enabled (following instructions from `pipx completions`).  This works fine in Bash, but in Xonsh completion for regular `pip` is used instead.

## xonfig

<details>

```
$ xonfig
+------------------+---------------------+
| xonsh            | 0.9.17.dev2         |
| Git SHA          | 97a921c7            |
| Commit Date      | Apr 8 14:48:23 2020 |
| Python           | 3.6.9               |
| PLY              | 3.11                |
| have readline    | True                |
| prompt toolkit   | 2.0.10              |
| shell type       | prompt_toolkit2     |
| pygments         | 2.6.1               |
| on posix         | True                |
| on linux         | True                |
| distro           | ubuntu              |
| on darwin        | False               |
| on windows       | False               |
| on cygwin        | False               |
| on msys2         | False               |
| is superuser     | False               |
| default encoding | utf-8               |
| xonsh encoding   | utf-8               |
| encoding errors  | surrogateescape     |
+------------------+---------------------+
```

</details>

## Expected Behavior
`pipx` completion to be used

## Current Behavior
`pip` completion is used.

## Steps to Reproduce

* Install `pipx`
* Try `pipx <tab>`.  Notice output is the same as for `pip <tab>`

## Possible fix
Adding the following (adds a trailing `\b` to the regex) to my `.xonshrc` seems to fix it:

```py
# Temp fix for `pipx` completion
import xonsh.completers.pip
import re
xonsh.completers.pip.PIP_RE = re.compile(r"\bx?pip(?:\d|\.)*\b")
```

Did some quick tests with: `pip <tab>`, `pip2 <tab>`, `pip3 <tab>`, `pipx <tab>`, `pip.3 <tab>`, `pip3.6 <tab>` and all seemed to give the expected results.
Also tested `xpip <tab>`.  This seems to (try to) use completion but I'm getting an error here (somewhere in the jedi code, probably unrelated).

**Comment by gforsyth** [comment]
Thanks for reporting, @sevens-ef -- the `pip` completer is a bit over-aggressive in it's firing triggers.  I can confirm this buggy behavior.

**Comment by sevens-ef** [comment]
Yeah, though the lookahead in the regex suggests it _should_ be more limited.  But in its current form it just matches anything that starts with `(x)pip` instead :)

**Comment by gforsyth** [comment]
@sevens-ef -- I have a fix in #3527 if you want to take a look

**Comment by sevens-ef** [comment]
Thanks.  Installed it locally and a quick test seems fine.  Will keep this as my main Xonsh install for now and will add a new comment here if I run into any problems.
